File: app_manager/appmanager/src/main/java/app/editors/manager/ui/activities/main/WebViewerActivity.java
package app.editors.manager.ui.activities.main;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.util.Log;

import androidx.fragment.app.Fragment;

import app.editors.manager.R;
import app.editors.manager.mvp.models.explorer.File;
import app.editors.manager.ui.activities.base.BaseAppActivity;
import app.editors.manager.ui.fragments.main.WebViewerFragment;


public class WebViewerActivity extends BaseAppActivity {

    public static final String TAG = WebViewerActivity.class.getSimpleName();
    public static final String TAG_VIEWER_FAIL = "TAG_VIEWER_FAIL";
    private static final String TAG_FILE = "TAG_FILE";

    private Thread.UncaughtExceptionHandler mUncaughtExceptionHandler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_viewer_web);
        init(savedInstanceState);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig){
        super.onConfigurationChanged(newConfig);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        killSelf();
    }

    private void init(final Bundle savedInstanceState) {
        initException();
        if (savedInstanceState == null) {
            final File file = (File) getIntent().getSerializableExtra(TAG_FILE);
            showFragment(WebViewerFragment.newInstance(file), null);
        }
    }

    private void initException() {
        mUncaughtExceptionHandler = Thread.getDefaultUncaughtExceptionHandler();
        Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
            Log.d(TAG, "ID: " + t.getId() + "; NAME: " + t.getName());
            final Intent intent = new Intent();
            setResult(Activity.RESULT_CANCELED, intent);
            mUncaughtExceptionHandler.uncaughtException(t, e);
        });
    }

    private static Intent getActivityIntent(final Context context) {
        final Intent intent = new Intent(context, WebViewerActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT);
        intent.addFlags(Intent.FLAG_ACTIVITY_MULTIPLE_TASK);
        return intent;
    }

    public static void show(final Fragment fragment, final File file) {
        final Intent intent = getActivityIntent(fragment.getContext());
        intent.putExtra(WebViewerActivity.TAG_FILE, file);
        fragment.startActivityForResult(intent, REQUEST_ACTIVITY_WEB_VIEWER);
    }

    public static void show(final Activity activity, final File file) {
        final Intent intent = getActivityIntent(activity);
        intent.putExtra(WebViewerActivity.TAG_FILE, file);
        activity.startActivityForResult(intent, REQUEST_ACTIVITY_WEB_VIEWER);
    }

}

File: app_manager/appmanager/src/main/java/app/editors/manager/ui/activities/main/WebViewerActivity.kt
package app.editors.manager.ui.activities.main

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import app.editors.manager.databinding.ActivityViewerWebBinding
import app.documents.core.network.manager.models.explorer.CloudFile
import app.editors.manager.ui.activities.base.BaseAppActivity
import app.editors.manager.ui.fragments.main.WebViewerFragment
import lib.toolkit.base.managers.utils.getSerializable

class WebViewerActivity : BaseAppActivity() {

    private var viewBinding: ActivityViewerWebBinding? = null
    private var uncaughtExceptionHandler: Thread.UncaughtExceptionHandler? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBinding = ActivityViewerWebBinding.inflate(layoutInflater)
        setContentView(viewBinding?.root)
        init(savedInstanceState)
    }

    override fun onDestroy() {
        super.onDestroy()
        viewBinding = null
        killSelf()
    }

    private fun init(savedInstanceState: Bundle?) {
        initException()
        savedInstanceState ?: run {
            showFragment(WebViewerFragment.newInstance(
                file = intent.getSerializable(TAG_FILE, CloudFile::class.java),
                isEditMode = intent.getBooleanExtra(TAG_IS_EDIT, false)
            ), null)
        }
    }

    private fun initException() {
        uncaughtExceptionHandler = Thread.getDefaultUncaughtExceptionHandler()
        Thread.setDefaultUncaughtExceptionHandler { thread: Thread, throwable: Throwable? ->
            Log.d(TAG, "ID: " + thread.id + "; NAME: " + thread.name)
            setResult(RESULT_CANCELED, Intent())
            uncaughtExceptionHandler?.uncaughtException(thread, throwable ?: Throwable())
        }
    }

    companion object {

        val TAG: String = WebViewerActivity::class.java.simpleName
        const val TAG_VIEWER_FAIL = "TAG_VIEWER_FAIL"
        const val TAG_FILE = "TAG_FILE"
        const val TAG_IS_EDIT = "TAG_IS_EDIT"

        fun getActivityIntent(context: Context, file: CloudFile, isEditMode: Boolean) =
            Intent(context, WebViewerActivity::class.java).apply {
                putExtra(TAG_FILE, file)
                putExtra(TAG_IS_EDIT, isEditMode)
                addFlags(Intent.FLAG_ACTIVITY_NEW_DOCUMENT)
                addFlags(Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
            }
    }
}